src/modinv32_impl.h

```diff
@@ -30,6 +66,17 @@ static void secp256k1_modinv32_normalize_30(secp256k1_modinv32_signed30 *r, int3
             r5 = r->v[5], r6 = r->v[6], r7 = r->v[7], r8 = r->v[8];
     int32_t cond_add, cond_negate;
 
+#ifdef VERIFY
+    /* Verify that all limbs are in range (-2^30,2^30). */
+    int i;
+    for (i = 0; i < 9; ++i) {
+        VERIFY_CHECK(r->v[i] >= -M30);
+        VERIFY_CHECK(r->v[i] <= M30);
```

Comment:
(-2^30,2^30) or [-2^30,2^30]?

Reply:
(-2^20,2^30) is correct. It's even stronger in practice I believe, and the bottom 8 limbs are in [0,2^30), and the top limb is in [-2^18,2^17) or so. Note that `_normalize` is only called on the `d` variable, and `d` is only ever updated through `_update_de` (which produces 8 output limbs masked by M30, and the range of the top limb can be reasoned about through the range of `d` itself).

Reply:
Also, note that M30 is 0x3fffffff, so it's actually checking (-2^30,2^30).